sdk/azidentity/ci.yml

```diff
@@ -39,3 +39,6 @@ extends:
             GenerateVMJobs: true
             Path: sdk/azidentity/managed-identity-matrix.json
```

Comment:
I'm not sure I understand what this actually buys us. Isn't it equivalent?

Reply:
@weshaggard the change to support hardcoded os types had us add filters that don't match the hardcoded pool name for identity (https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/pipelines/templates/jobs/generate-job-matrix.yml#L48-L57), as the filters expect the env var format instead. We could update the filters (for storage and identity pool names) or also consider adding another env var to image.yml that matches e.g. `LINUXAZIDENTITYPOOL`.

Reply:
OK so it is the fact that those are looking for the env names in the matrix to determine. This seems like it might be a little order dependent and magical but I'm OK with it for now. We probably should document that somewhere to state what string matching we are dependent on in the matrix.

Reply:
We have some comments in the configs and the matrix documentation also specifies that we process filter/replace against the environment variable keys, not the resolved values.

Reply:
I think going forward I will make an update to image.yml for the repo, as that keeps things all in one tidy place.